# Exercício 1 da Aula 1

In [1]:
import pandas as pd


In [2]:
# lendo o conjunto de dados do diretório
data = pd.read_csv('/content/drive/MyDrive/DS/Dataset/kc_house_data.csv')

In [3]:
# mostra as 5 primeiras linhas do dataset
data.head()

id             date     price  ...     long  sqft_living15  sqft_lot15
0  7129300520  20141013T000000  221900.0  ... -122.257           1340        5650
1  6414100192  20141209T000000  538000.0  ... -122.319           1690        7639
2  5631500400  20150225T000000  180000.0  ... -122.233           2720        8062
3  2487200875  20141209T000000  604000.0  ... -122.393           1360        5000
4  1954400510  20150218T000000  510000.0  ... -122.045           1800        7503

[5 rows x 21 columns]

In [4]:
#mostra o tamanho do dataset
data.shape

(21613, 21)

In [5]:
# Mostra o nome das colunas
data.columns

Index(['id', 'date', 'price', 'bedrooms', 'bathrooms', 'sqft_living',
       'sqft_lot', 'floors', 'waterfront', 'view', 'condition', 'grade',
       'sqft_above', 'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode',
       'lat', 'long', 'sqft_living15', 'sqft_lot15'],
      dtype='object')

In [6]:
# Moatrar na tela os dados ordenados da coluna "Price"
data.sort_values('price')

id             date  ...  sqft_living15  sqft_lot15
1149   3421079032  20150217T000000  ...           1160       42882
15293    40000362  20140506T000000  ...           1700       10387
465    8658300340  20140523T000000  ...           1200        7500
16198  3028200080  20150324T000000  ...            860        9000
8274   3883800011  20141105T000000  ...           1140       11250
...           ...              ...  ...            ...         ...
1448   8907500070  20150413T000000  ...           4600       21750
4411   2470100110  20140804T000000  ...           3560       24345
9254   9208900037  20140919T000000  ...           4540       42730
3914   9808700762  20140611T000000  ...           3930       25449
7252   6762700020  20141013T000000  ...           3940        8800

[21613 rows x 21 columns]

In [7]:
# Moatrar na tela os dados ordenados da coluna "Price" do maior para o menor ṕelo e pela coluna price
data[['id','price']].sort_values('price', ascending = False)

id      price
7252   6762700020  7700000.0
3914   9808700762  7062500.0
9254   9208900037  6885000.0
4411   2470100110  5570000.0
1448   8907500070  5350000.0
...           ...        ...
8274   3883800011    82000.0
16198  3028200080    81000.0
465    8658300340    80000.0
15293    40000362    78000.0
1149   3421079032    75000.0

[21613 rows x 2 columns]

# Retirando itens duplicados

In [19]:
data['id'].duplicated().value_counts() # verifica se existem itens duplicados

False    21436
True       177
Name: id, dtype: int64

In [21]:

data.drop_duplicates(subset='id', keep='last', inplace=True) # retira os itens duplicados

# Perguntas do CEO

1) Quantas casas estão disponíveis para compra? Existem 21.436 casas disponíveis

In [22]:
data.shape

(21436, 21)

2) Quantos atributos as casa possuem? 21 nomes correspondem aos atributos das casas

In [9]:
data.columns

Index(['id', 'date', 'price', 'bedrooms', 'bathrooms', 'sqft_living',
       'sqft_lot', 'floors', 'waterfront', 'view', 'condition', 'grade',
       'sqft_above', 'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode',
       'lat', 'long', 'sqft_living15', 'sqft_lot15'],
      dtype='object')

3) Quais são os atributos das casas? 'id', 'date', 'price', 'bedrooms', 'bathrooms', 'sqft_living',
       'sqft_lot', 'floors', 'waterfront', 'view', 'condition', 'grade',
       'sqft_above', 'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode',
       'lat', 'long', 'sqft_living15', 'sqft_lot15'

4) Qual a casa mais cara? É a casa cujo id = 6762700020


In [10]:
data[['id', 'price']].sort_values('price', ascending = False )


id      price
7252   6762700020  7700000.0
3914   9808700762  7062500.0
9254   9208900037  6885000.0
4411   2470100110  5570000.0
1448   8907500070  5350000.0
...           ...        ...
8274   3883800011    82000.0
16198  3028200080    81000.0
465    8658300340    80000.0
15293    40000362    78000.0
1149   3421079032    75000.0

[21613 rows x 2 columns]

5) Qual a casa com o maior número de quartos? A casa Id =2402100895 possue 33 quartos

In [11]:
data[['id','bedrooms']].sort_values('bedrooms', ascending = False)

id  bedrooms
15870  2402100895        33
8757   1773100755        11
15161  5566100170        10
13314   627300145        10
19254  8812401450        10
...           ...       ...
19452  3980300371         0
8484   2310060040         0
875    6306400140         0
8477   2569500210         0
9773   3374500520         0

[21613 rows x 2 columns]

6) Qual a soma total de quartos do conjunto de dados? Existem 72273 quartos


In [23]:
data['bedrooms'].sum()

72273

7) Quantas casas possuem 2 banheiros? Existem 1913 casas com 2 quartos

In [24]:

data.query("bathrooms == 2")["bathrooms"].count()

1913

8) Qual o preço médio de todas as casas no conjunto de dados? Valor médio de 541649.962726255

In [25]:
data['price'].mean()

541649.962726255

9) Qual o preço médio de casas com 2 banheiros?

In [26]:
data.query("bathrooms == 2")[['bathrooms','price']].mean()

bathrooms         2.000000
price        459307.013591
dtype: float64

10) Qual o preço mínimo entre as casa com 3 quartos? O menor valoe é de 89000

In [27]:
data.query("bedrooms == 3").sort_values('price').head()

id             date  ...  sqft_living15  sqft_lot15
5866   9320900420  20141014T000000  ...            900        3404
18939  4239400300  20141129T000000  ...            980        3154
14283  1788800610  20140902T000000  ...            840        9450
14525  4311700120  20150317T000000  ...            940       11193
17197  5111400086  20140512T000000  ...           1820      217800

[5 rows x 21 columns]

11) Quantas casas possuem mais de 300 metros quadrados (m²)?

In [33]:
# vamos trocar a medida 'pés' por 'metros', utilizando a função "change"
def change(area):
    return area/(3.28084**2)

In [36]:

data['sqm_living'] = change(data['sqft_living'][:]) # convertendo as medidas para metros quadrados
data.loc[data['sqft_living']>300, 'sqft_living'].value_counts().sum()

21435

12) Questão 12: Quantas casas possuem mais de 2 andares?


In [37]:
data.loc[data['floors']> 2,'floors'].value_counts().sum()

780

13) Quantas casas tem vista para o mar?

In [39]:
data.loc[data['waterfront'] != 0, 'waterfront'].value_counts().item()

163

14) Quantas casas possuem vista para o mar e 3 quartos?

In [42]:
data.loc[(data['waterfront'] != 0) & (data['bedrooms'] == 3), 'id'].value_counts().sum()


64

15) Quantas casas possuem mais de 300 m² de sala de estar e mais de 2 banheiros?

In [45]:
data.loc[(data['sqm_living'] > 300) & (data['bathrooms'] > 2), 'id'].value_counts().sum()


2194